app/services/change_offer.rb

```diff
@@ -1,16 +1,28 @@
 class ChangeOffer
   include ActiveModel::Validations
 
-  def initialize(actor:, application_choice:, course_option_id:)
+  attr_reader :application_choice, :course_option_id
+
+  validates_each :course_option_id do |record, attr, value|
+    record.errors.add(attr, :no_change) if value == record.application_choice.offered_option.id
```

Comment:
What if they're only changing the conditions? It doesn't feel fair to reject that change because they also included the course option which hadn't changed.